android/java/org/chromium/chrome/browser/ntp_background_images/util/NTPUtil.java

```diff
@@ -344,6 +352,56 @@ public static Bitmap getWallpaperBitmap(NTPImage ntpImage, int layoutWidth, int
         return bitmapWithGradient;
     }
 
+    public static Bitmap getTopSiteBitmap(String iconPath) {
+        Context mContext = ContextUtils.getApplicationContext();
+        InputStream inputStream = null;
+        Bitmap topSiteIcon = null;
+        try {
+            Uri imageFileUri = Uri.parse("file://"+ iconPath);
+            inputStream = mContext.getContentResolver().openInputStream(imageFileUri);
```

Comment:
do we execute that on a correct thread? I mean don't we block UI thread on that place and have a StrictMode warning?

Reply:
@SergeyZhukovsky this function is called from a separate thread from BraveNewTabPageView.